index.d.ts

```diff
@@ -15,19 +15,802 @@
+     * Should return an object (literal) with key value pairs.
+     * Can contain nested objects, but beware that too deep nesting can have
+     * a negative impact on performance
+     *
+     * @example
+     * ```js
+     * state() {
+     *  return {
+     *    items: [],
+     *    color: 'red',
+     *    alpha: 0.1
+     *  }
+     * }
+     * ```
+     */
+    state?: (this: PropsDefinition<P>) => S;
+    /**
+     * Methods for abstracting more complex business logic into separate function
+     */
+    methods?: M & ComponentContext<P, S, M, C>
+    /**
+     * Hooking into Lifecycle events
+     */
+    hooks?: Hooks & ComponentContext<P, S, M, C>
+    /**
+     * Tapping into user input
+     */
+    input?: Input & ComponentContext<P, S, M, C>
+    /**
+     * Computed properties
+     */
+    computed?: C & ComponentContext<P, S, M, C>
+    /**
+     * Watchers for changes to state variables, props or computed properties
+     */
+    watch?: W & ComponentContext<P, S, M, C>
+  }
+
+  interface ApplicationConfig<P extends Props, S, M, C, W> extends ComponentConfig<P, S, M, C, W> {
+    /**
+     * Routes definition
+     *
+     * @example
+     *
+     * ```js
+     * routes: [
+     *  { path: '/', component: Home },
+     *  { path: '/details', component: Details },
+     *  { path: '/account', component: Account },
+     * ]
+     * ```
+     */
+    routes?: Route[]
+  }
 
-export default Blits
+
+  interface Transition {
+    /**
+     * Name of the prop to transition (i.e. 'x', 'y', 'alpha', 'color')
+     */
+    prop: string,
+    /**
+     * Value the prop should transition to (i.e. 0, 100, '#223388')
+     */
+    value: any,
+    /**
+     * Duration of the transition in milliseconds, defaults to 300
+     */
+    duration?: number,
+    /**
+     * Easing function to apply to the transition
+     */
+    easing?: string,
+    /**
+     * Delay before the transition starts in milliseconds
+     */
+    delay?: number
+  }
+
+  interface Before {
+    /**
+    * Name of the prop to set before the transition starts
+    */
+    prop: string,
+    /**
+     * Value the prop to set before the transition starts
+     */
+    value: any,
+  }
+
+  interface RouteTransition {
+    /**
+     * Setting or Array of Settings before new view enters into the router view
+     */
+    before: Before | Before[],
+    /**
+     * Transition or Array of Transitions for new view to enters into the router view
+     */
+    in: Transition | Transition[],
+    /**
+     * Transition or Array of Transitions for old view to leave the router view
+     */
+    out: Transition | Transition[],
+  }
+
+  type RouteTransitionFunction = (previousRoute: Route, currentRoute: Route) => RequireAtLeastOne<RouteTransition>
+
+  interface RouteAnnounce {
+    /**
+     * Message to be announced
+     */
+    message: String,
+    /**
+     * Politeness level
+     *
+     * Defaults to 'off'
+     */
+    politeness?: 'off' | 'polite' | 'assertive'
```

Comment:
what does this do?

Reply:
it's a property used in text to speech to indicate critical notifications that require immediate attention